simulator-doc-extended.txt

```diff
@@ -0,0 +1,353 @@
+	the switchIn event.
+
+	examples: [volatile] Encore, [side condition] Reflect,
+	          [volatile] Substitute
+
+restart(pokemon) [on status, on volatile]
+restart(side) [on side condition]
+restart() [on weather, on field effect]
+	Fired when trying to induce an effect that's already active.
+
+	Useful for effects that can have multiple "layers".
+
+	examples: [side condition] Toxic Spikes, [side condition] Pursuit,
+	          [volatile] Stockpile
+
+end(pokemon) [on status, on volatile, on item, on ability]
+end(side) [on side condition]
+end() [on weather, on field effect]
+	Fired when an effect ends naturally, usually because its duration ends.
+	Return true to stop an effect from ending.
+
+	Most ways an effect can end are "natural". The biggest exception is
+	when a pokemon swiches out. In general, if you would receive a message
+	when the effect ends, the end event would be fired.
+
+	NOTE: NOT fired when a pokemon switches out (see switchOut).
+
+	NOTE: If you return true when an event expires from its duration counter
+	      reaching 0, it will never expire unless you reset its duration. To
+	      extend an effect by one turn, simply set its duration counter to 1
+	      before returning true.
+
+	Useful for giving messages that an effect has ended.
+
+	examples: [volatile] Encore, [side condition] Reflect,
+	          [volatile] Substitute
+
+duration(pokemon, source) [on status, on volatile]
+duration(side) [on side condition]
+duration() [on weather, on field effect]
+	Fired while calculating an effect's duration. Returns the duration in
+	turns, including the current one.
+
+	source is the pokemon that inflicted the effect, or null if no such
+	pokemon did. For instance, if Pikachu uses Toxic, source is Pikachu,
+	but if you are poisoned by Toxic Spikes, source is null.
+
+	For instance, Protect has a duration of 1 turn, and Wish has a duration
+	of 2 turns. Return 0 to have it never expire.
+
+	The effect's residual event is not fired when it expires - instead, its
+	end event is fired when its residual event would otherwise have fired.
+
+	Do not use this for effects that don't expire during a residual. For
+	instance, sleep expires through a sleep counter that decrements every
+	beforeMove event, not every residual event.
+
+	examples: [weather] Rain Dance, [side condition] Reflect
+
+residual(pokemon) [on status, on volatile, on ability, on item]
+residual(side) [on side condition]
+residual() [on weather, on field effect]
+	Fired at the end of each turn, but before fainted pokemon are switched in.
+
+	Useful for whatever end-of-turn effects this may have.
+
+	The duration counter of each effect is decremented automatically before
+	this event is fired (see duration for details).
+
+	examples: [volatile] Ghost-type Curse, [weather] Sandstorm
+
+
+=== DETAILED USAGE NOTES ===
+
+In Pokémon Showdown, as well as in Pokémon Lab (https://pokemonlab.com/) and
+DougJustDoug's framework, everything in Pokémon is an effect.
```

Comment:
What's this framework? I found this https://www.smogon.com/forums/threads/damage-calculator.65197/, but I can't be sure that's it.

Reply:
Note: I've already updated the content of this section.

Reply:
iirc he never ended up publishing it